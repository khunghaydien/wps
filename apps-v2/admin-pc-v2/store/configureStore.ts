import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer from '../reducers';

const PRODUCTION = process.env.NODE_ENV === 'production';

const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
        traceLimit: 25,
      })
    : compose;

export default function configureStore(initialState: any) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../reducers', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../reducers').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
