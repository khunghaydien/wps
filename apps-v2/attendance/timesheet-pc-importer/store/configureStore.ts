/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
// 本来は使いたくないが
// Global Container の社員切替が使用しているので
// しかたなく導入している
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.TEST === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middleWares: Middleware[] = [
    // 本来は使いたくないが
    // Global Container の社員切替が使用しているので
    // しかたなく導入している
    thunkMiddleware,
  ];
  if (!PRODUCTION && !TEST) {
    middleWares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middleWares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
