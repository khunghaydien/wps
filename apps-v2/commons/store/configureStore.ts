import { AnyAction, applyMiddleware, createStore, Store } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../reducers';

const createStoreWithMiddleware = applyMiddleware(
  thunkMiddleware,
  logger
)(createStore);

export default function configureStore(
  initialState?: State
): Store<State, AnyAction> {
  const store = createStoreWithMiddleware(rootReducer, initialState);

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      /* eslint-disable  @typescript-eslint/no-var-requires */
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const nextRootReducer = require('../reducers');

      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
}
