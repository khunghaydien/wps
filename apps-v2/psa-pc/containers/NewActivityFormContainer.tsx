import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AnyAction, bindActionCreators } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

import { Formik } from 'formik';
import cloneDeep from 'lodash/cloneDeep';

import newActivityFormSchema from '@psa/schema/NewActivityForm';

import { catchBusinessError } from '@commons/actions/app';

import { State } from '@psa/modules';

import { saveActivity } from '@psa/action-dispatchers/Activity';
import { fetchProject } from '@psa/action-dispatchers/Project';
import { hideDialog } from '@psa/action-dispatchers/PSA';

import NewActivityForm from '@psa/components/Dialog/NewActivityForm';

import { getManagerList } from './ProjectDetailFormContainer';

const NewActivityFormContainer = () => {
  const dispatch = useDispatch() as ThunkDispatch<unknown, void, AnyAction>;

  const companyId = useSelector((state: State) => state.userSetting.companyId);
  const managerListOption = useSelector((state: State) =>
    getManagerList(state.entities)
  );
  const selectedProject = useSelector(
    (state: State) => state.entities.psa.project.project
  );
  const selectedActivity = useSelector(
    (state: State) => state.entities.psa.activity.activity
  );
  const useExistingJobCode = useSelector(
    (state: State) => state.entities.psa.setting.useExistingJobCode
  );
  const enableProgressCheck = useSelector(
    (state: State) => state.entities.psa.setting.enableProgressCheck
  );

  const generateInitialValues = () => ({
    code: '',
    description: '',
    jobCode: useExistingJobCode ? selectedProject.jobCode : '',
    jobId: useExistingJobCode ? selectedProject.jobId : '',
    leadBaseId: '',
    plannedEndDate: selectedProject
      ? selectedProject.endDate
      : selectedActivity.projectEndDate,
    plannedStartDate: selectedProject
      ? selectedProject.startDate
      : selectedActivity.projectStartDate,
    status: '',
    title: '',
    useExistingJobCode: useExistingJobCode,
  });

  const handleSubmit = (values) => {
    const activity: any = cloneDeep(values);
    activity.projectId = selectedProject.projectId;

    // remove ui values
    if (!useExistingJobCode) {
      delete activity.jobId;
    }
    delete activity.jobCode;
    delete activity.useExistingJobCode;

    dispatch(saveActivity(activity));
  };

  const Actions = bindActionCreators(
    {
      catchBusinessError,
      fetchProject,
      hideDialog,
    },
    dispatch
  );

  return (
    <Formik
      enableReinitialize
      initialValues={generateInitialValues()}
      validationSchema={newActivityFormSchema}
      onSubmit={handleSubmit}
    >
      {(props) => {
        return (
          <NewActivityForm
            catchBusinessError={Actions.catchBusinessError}
            companyId={companyId}
            dirty={props.dirty}
            errors={props.errors}
            handleSubmit={props.handleSubmit}
            hideDialog={Actions.hideDialog}
            managerListOption={managerListOption}
            selectedActivity={selectedActivity}
            selectedProject={selectedProject}
            setFieldValue={props.setFieldValue}
            touched={props.touched}
            useExistingJobCode={useExistingJobCode}
            values={props.values}
            enableProgressCheck={enableProgressCheck}
          />
        );
      }}
    </Formik>
  );
};

export default NewActivityFormContainer;
