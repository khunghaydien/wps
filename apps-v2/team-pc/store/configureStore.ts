/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
