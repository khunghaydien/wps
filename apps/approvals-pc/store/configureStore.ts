/* eslint-disable global-require, import/newline-after-import */
import {
  AnyAction,
  applyMiddleware,
  compose,
  createStore,
  Middleware,
  Store,
} from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';
import { middlewares as requestCountsMiddlewares } from '../modules/ui/requestCounts';

const PRODUCTION = process.env.NODE_ENV === 'production';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(
  initialState?: State
): Store<State, AnyAction> {
  let middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  middlewares = [...middlewares, ...requestCountsMiddlewares];

  const store = createStore(
    rootReducer,
    initialState,
    // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
