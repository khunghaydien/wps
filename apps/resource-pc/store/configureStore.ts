/* eslint-disable global-require, import/newline-after-import */
import { AnyAction, applyMiddleware, compose, createStore, Store } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '@resource/modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(
  initialState?: State
): Store<State, AnyAction> {
  const middlewares = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    // @ts-ignore
    middlewares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState,
    // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
