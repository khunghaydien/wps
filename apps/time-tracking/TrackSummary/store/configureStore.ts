/* eslint-disable global-require, import/newline-after-import */
import {
  AnyAction,
  applyMiddleware,
  compose,
  createStore,
  Middleware,
  Store,
  StoreEnhancer,
} from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

interface ExtendedWindow extends Window {
  __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
}
declare let window: ExtendedWindow;

/* eslint-disable no-underscore-dangle */
function composeEnhancers(
  name = 'TrackSummary'
): (enhancer: StoreEnhancer) => StoreEnhancer {
  return !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ name })
    : compose;
}

type Params = {
  name?: string;
  initialState?: Partial<State>;
};

export default function configureStore({
  name,
  initialState,
}: Params = {}): Store<typeof initialState, AnyAction> {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    middlewares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState,
    // Redux DevTools Extension
    composeEnhancers(name)(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
