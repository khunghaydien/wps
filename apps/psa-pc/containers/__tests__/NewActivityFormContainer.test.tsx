import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import NewActivityFormContainer from '../NewActivityFormContainer';

const mockStore = configureStore([thunk]);

const state = {
  entities: {
    psa: {
      project: {
        project: {},
      },
      activity: {
        activity: {},
      },
      setting: {
        useExistingJobCode: '',
      },
    },
  },
  userSetting: {
    companyId: '',
  },
};

// FIXME: Thrown 'Attempt to call useDialog out of dialog context. Make sure your app is rendered in DialogProvider.'
describe.skip('NewActivityFormContainer Test', () => {
  let store;
  let component;
  beforeEach(async () => {
    store = mockStore(state);
    await renderer.act(() => {
      component = renderer.create(
        <Provider store={store}>
          <NewActivityFormContainer />
        </Provider>
      );
    });
  });

  it('it renders without crashing', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
