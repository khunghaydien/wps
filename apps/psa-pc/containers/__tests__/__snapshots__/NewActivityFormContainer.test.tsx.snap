// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewActivityFormContainer Test it renders without crashing 1`] = `
<div
  className="commons-dialog-frame__wrapper"
  style={
    Object {
      "zIndex": 5100000,
    }
  }
>
  <div
    className="commons-dialog-frame__overlay"
    onClick={[Function]}
  />
  <div
    className="commons-dialog-frame slds ts-psa__new-activity-form__dialog-frame react-draggable"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    role="dialog"
    style={
      Object {
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  >
    <div
      className="slds-grid slds-grid--vertical-align-center commons-dialog-frame__header commons-dialog-frame__header__cursor"
    >
      <div
        className="slds-grow commons-dialog-frame__header-title"
      >
        New {$Psa_Clbl_Activity}
      </div>
    </div>
    <div
      className="commons-dialog-frame__wrap"
    >
      <div
        className="commons-dialog-frame__contents"
      >
        <div
          className="ts-psa__new-activity-form__inner"
        >
          <div
            className="ts-psa__new-activity-form__sub-header"
          >
            <div
              className="ts-psa__new-activity-form__sub-header__title"
              data-testid="ts-psa__new-activity-form__sub-header__title"
            >
              <span
                className="ts-psa__new-activity-form__sub-header__title-header"
              >
                Project Title
              </span>
              :
              <span
                className="ts-psa__new-activity-form__sub-header__title-header-body"
              />
            </div>
            <div
              className="ts-psa__new-activity-form__sub-header__duration"
              data-testid="ts-psa__new-activity-form__sub-header--duration"
            >
              <span
                className="ts-psa__new-activity-form__sub-header__duration-header"
              >
                 
                Duration
              </span>
              : 
              
                  -
            
            </div>
          </div>
          <div
            className="ts-psa__new-activity-form"
          >
            <div
              className="ts-psa__new-activity-form__form-field-container"
            >
              <div
                className="ts-psa__form-field  "
                data-testid="ts-psa__new-activity-form--title"
              >
                <div
                  className="ts-psa__form-field__label"
                >
                  <span
                    className="is-required"
                  >
                    *
                  </span>
                   
                  {$Psa_Clbl_Activity} Title
                </div>
                <input
                  className="ts-text-field slds-input"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="ts-psa__form-field  "
                data-testid="ts-psa__new-activity-form--code"
              >
                <div
                  className="ts-psa__form-field__label"
                >
                  <span
                    className="is-required"
                  >
                    *
                  </span>
                   
                  {$Psa_Clbl_Activity} Code
                </div>
                <input
                  className="ts-text-field slds-input"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder={null}
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="ts-psa__new-activity-form__form-field-container"
            >
              <div
                className="ts-psa__form-field  "
                data-testid="ts-psa__new-activity-form--start-date"
              >
                <div
                  className="ts-psa__form-field__label"
                >
                  <span
                    className="is-required"
                  >
                    *
                  </span>
                   
                  Start Date
                </div>
                <div
                  className="ts-psa__date-field__container"
                >
                  <div
                    className="commons-fields-date-field"
                    data-testid=""
                  >
                    <div>
                      <div
                        className="react-datepicker-wrapper"
                      >
                        <div
                          className="react-datepicker__input-container"
                        >
                          <input
                            className="slds-input"
                            disabled={false}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Select Start Date"
                            readOnly={false}
                            required={false}
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="ts-psa__date-field__calendar-icon-wrapper"
                  >
                    <test-file-stub
                      className="ts-psa__date-field__calendar-icon-body"
                    />
                  </div>
                </div>
              </div>
              <div
                className="ts-psa__form-field  "
                data-testid="ts-psa__new-activity-form--end-date"
              >
                <div
                  className="ts-psa__form-field__label"
                >
                  <span
                    className="is-required"
                  >
                    *
                  </span>
                   
                  End Date
                </div>
                <div
                  className="ts-psa__date-field__container"
                >
                  <div
                    className="commons-fields-date-field"
                    data-testid=""
                  >
                    <div>
                      <div
                        className="react-datepicker-wrapper"
                      >
                        <div
                          className="react-datepicker__input-container"
                        >
                          <input
                            className="slds-input"
                            disabled={false}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Select End Date"
                            readOnly={false}
                            required={false}
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="ts-psa__date-field__calendar-icon-wrapper"
                  >
                    <test-file-stub
                      className="ts-psa__date-field__calendar-icon-body"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="commons-dialog-frame__footer"
    >
      <div
        className="slds-grid commons-dialog-frame-footer"
      >
        <div
          className="slds-grow commons-dialog-frame-footer__sub"
        />
        <div
          className="commons-dialog-frame-footer__inner"
        >
          <button
            className="ts-button ts-button--default"
            data-testid="ts-psa__new-activity-form--cancel"
            disabled={false}
            onClick={[Function]}
            type="button"
            value=""
          >
            <div
              className="ts-button__contents"
            >
              <span
                className="ts-button__text"
              >
                Cancel
              </span>
            </div>
          </button>
          <button
            className="ts-button ts-button--primary"
            data-testid="ts-psa__new-activity-form--save"
            disabled={true}
            onClick={[Function]}
            type="button"
            value=""
          >
            <div
              className="ts-button__contents"
            >
              <span
                className="ts-button__text"
              >
                Save
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
